jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessorTest.java

```diff
@@ -838,6 +874,58 @@ public void testGetJavaContainerBuilderWithBaseImage_java12NoBaseImage()
     }
   }
 
+  @Test
+  public void testGetValidPlatformsSet() throws InvalidPlatformConfigurationException {
+    Mockito.<List<? extends PlatformConfiguration>>when(rawConfiguration.getPlatforms())
+        .thenReturn(Arrays.asList(new TestPlatformConfiguration("testArchitecture", "testOs")));
+
+    Assert.assertEquals(
+        ImmutableSet.of(new Platform("testArchitecture", "testOs")),
+        PluginConfigurationProcessor.getPlatformsSet(rawConfiguration));
+  }
+
+  @Test
+  public void testGetPlatformsSet_architectureMissing()
+      throws InvalidPlatformConfigurationException {
+    TestPlatformConfiguration platform = new TestPlatformConfiguration(null, "testOs");
+    Mockito.<List<? extends PlatformConfiguration>>when(rawConfiguration.getPlatforms())
+        .thenReturn(Arrays.asList(platform));
+
+    try {
+      PluginConfigurationProcessor.getPlatformsSet(rawConfiguration);
+      Assert.fail();
+    } catch (InvalidPlatformConfigurationException ex) {
+      String message =
+          "architecture="
+              + platform.getArchitectureName().orElse("<missing>")
+              + ", os="
+              + platform.getOsName().orElse("<missing>");
+      Assert.assertEquals(
+          "platform configuration is missing an architecture value", ex.getMessage());
+      Assert.assertEquals(message, ex.getInvalidPlatform());
+    }
+  }
+
+  @Test
+  public void testGetPlatformsSet_osMissing() throws InvalidPlatformConfigurationException {
+    TestPlatformConfiguration platform = new TestPlatformConfiguration("testArchitecture", null);
+    Mockito.<List<? extends PlatformConfiguration>>when(rawConfiguration.getPlatforms())
+        .thenReturn(Arrays.asList(platform));
+
+    try {
+      PluginConfigurationProcessor.getPlatformsSet(rawConfiguration);
+      Assert.fail();
+    } catch (InvalidPlatformConfigurationException ex) {
+      String message =
+          "architecture="
+              + platform.getArchitectureName().orElse("<missing>")
+              + ", os="
+              + platform.getOsName().orElse("<missing>");
```

Comment:
`message = "architecture=testArchitecture, os=<missing>"` since we are verifying the return value of `ex.getInvalidPlatform()`.